File: application/src/main/java/de/pbauerochse/worklogviewer/domain/timerangeprovider/LastWeekTimerangeProvider.java
package de.pbauerochse.worklogviewer.domain.timerangeprovider;

import de.pbauerochse.worklogviewer.domain.ReportTimerange;
import de.pbauerochse.worklogviewer.report.TimeRange;

public class LastWeekTimerangeProvider extends BaseTimerangeProvider {

    @Override
    public TimeRange getTimeRange() {
        return TimeRange.lastWeek();
    }

    @Override
    public ReportTimerange getReportTimerange() {
        return ReportTimerange.LAST_WEEK;
    }

}

File: application/src/main/java/de/pbauerochse/worklogviewer/timerange/LastWeekTimerangeProvider.kt
package de.pbauerochse.worklogviewer.timerange

import de.pbauerochse.worklogviewer.timereport.TimeRange
import de.pbauerochse.worklogviewer.util.FormattingUtil.getFormatted
import java.time.LocalDate

/**
 * The previous week as [TimerangeProvider]
 */
internal object LastWeekTimerangeProvider : TimerangeProvider {
    override val label: String = getFormatted("timerange.lastweek")
    override val settingsKey: String = "LAST_WEEK"
    override val isComputed: Boolean = true
    override fun buildTimeRange(start: LocalDate?, end: LocalDate?): TimeRange {
        return TimeRange.lastWeek()
    }
}
